<?xml version="1.0"?>
<launch>
    <arg name="roomba"      default="roomba1" />
    <arg name="config_path" default="$(find place_recognition)/config" />

    <!-- rosbag -->
    <include file="$(find ros_utils)/launch/bag_player.launch" >
        <arg name="roomba"               value="$(arg roomba)" />
        <arg name="wait_time"            value="0.0" />
        <arg name="file_path"            value="$(env HOME)/bagfiles/experimental_file_20220630/" />
        <arg name="convertor_theta_s"    value="true" />
        <arg name="publish_objects_data" value="false" />
    </include>

    <!-- reference_data_creator -->
    <node pkg="place_recognition" type="reference_data_creator" name="reference_data_creator" output="screen" >
        <!-- remap -->
        <remap from="equ_in"  to="$(arg roomba)/equirectangular/image_raw" />
        <remap from="rgb_in"  to="$(arg roomba)/camera/color/image_rect_color" />
        <remap from="pose_in" to="$(arg roomba)/amcl_pose" />

        <!-- params -->
        <param name="FILE_PATH" value="$(find place_recognition)/record/$(arg roomba)" />
        <param name="IS_RECORD" value="false" />

        <!-- rosparams -->
        <rosparam command="load" file="$(arg config_path)/params/inpaintor/$(arg roomba).yaml" />
    </node>

</launch>
